src/main/java/com/exasol/adapter/document/documentfetcher/files/csv/CsvHeaderDetector.java

```diff
@@ -0,0 +1,113 @@
+package com.exasol.adapter.document.documentfetcher.files.csv;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.*;
+import java.util.logging.Logger;
+
+import com.exasol.adapter.document.documentfetcher.files.RemoteFile;
+
+import io.deephaven.csv.CsvSpecs;
+import io.deephaven.csv.parsers.DataType;
+import io.deephaven.csv.reading.CsvReader;
+import io.deephaven.csv.reading.CsvReader.ResultColumn;
+
+/**
+ * This class detects if a CSV file contains a header row or not.
+ */
+class CsvHeaderDetector {
+    private static final Logger LOG = Logger.getLogger(CsvHeaderDetector.class.getName());
+
+    private final RemoteFile remoteFile;
+
+    CsvHeaderDetector(final RemoteFile remoteFile) {
+        this.remoteFile = remoteFile;
+    }
+
+    boolean hasHeaderRow() {
```

Comment:
Would it make sense to return a more detailed result type here?
```java
public class HeaderStatus {
    public static HeaderStatus nonStringTypesInFirstRow(String  dataTypes) {
       return new HeaderStatus(false, "First row contains data with non-string type: " + dataTypes);
    }
 
    public static HeaderStatus identicalDataTypesInFistRows(String dataTypes) {
       return new HeaderStatus(false, "First and second row contain data with identical types: " + dataTypes);
    }

    public static HeaderStatus withHeaders() {
       return new HeaderStatus(true, "");
    }
   
    private final boolean verdict; 
    private final String reason;

    HeaderStatus(boolean verdict, String reason) {
      this.verdict = verdict; 
      this.reason = reason;
    } 
}
```

Reply:
Comment #2: Could the user also configure the auto-detection, e.g. by providing a flag telling whether the CSV file has headers? 

I anticipate the answer: "Yes, but we prefer to make auto-detection as simple as possible and any kind of configuration will fall back to simply asking the user to provide a EDML mapping file.", right?